
What is Exception?

An exception is an event, which occurs during the execution of a 
program that disrupts the normal flow of the program's instructions.


```

try:
   You do your operations here
   ......................
except ExceptionI:
   If there is ExceptionI, then execute this block.
except ExceptionII:
   If there is ExceptionII, then execute this block.
   ......................
else:
   If there is no exception then execute this block.


```


```

try:
   fh = open("testfile", "w")
   fh.write("This is my test file for exception handling!!")
except IOError:
   print ("Error: can\'t find file or read data")
else:
   print ("Written content in the file successfully")
   fh.close()


```


The try-finally Clause
------------------------

You can use a finally: block along with a try: block. The finally block is a 
place to put any code that must execute, whether the try-block 
raised an exception or not. The syntax of the try-finally statement is this âˆ’

```
try:
   fh = open("testfile", "w")
   fh.write("This is my test file for exception handling!!")
finally:
   print ("Error: can\'t find file or read data")

```


```

try:
   fh = open("testfile", "w")
   try:
      fh.write("This is my test file for exception handling!!")
   finally:
      print ("Going to close the file")
      fh.close()
except IOError:
   print ("Error: can\'t find file or read data")


```

















